AndroidRepository/RepositoryManager.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using static System.Net.WebRequestMethods;
+
+namespace AndroidRepository;
+
+public class RepositoryManager
+{
+	readonly HttpClient httpClient;
+	private readonly Dictionary<Type, XmlSerializer> serializerCache = new();
+
+	public const string GoogleAndroidRepositoryUrlBase = "https://dl.google.com/android/repository/";
+
+	public RepositoryManager(HttpClient? httpClient = null)
+	{
+		if (this.httpClient is null)
+		{
+			this.httpClient = new HttpClient();
+			this.httpClient.DefaultRequestHeaders.TryAddWithoutValidation("Accept", "text/html,application/xhtml+xml,application/xml");
+			this.httpClient.DefaultRequestHeaders.TryAddWithoutValidation("Accept-Encoding", "gzip, deflate");
+			this.httpClient.DefaultRequestHeaders.TryAddWithoutValidation("User-Agent", "Mozilla/5.0 (Windows NT 6.2; WOW64; rv:19.0) Gecko/20100101 Firefox/19.0");
+			this.httpClient.DefaultRequestHeaders.TryAddWithoutValidation("Accept-Charset", "ISO-8859-1");
+		}
+	}
+
+	public async Task<T> LoadUrlAsync<T>(string url)
+	{
+		var xml = await httpClient.GetStringAsync(url);
+
+		return await LoadXmlAsync<T>(xml);
+	}
+
+	public async Task<T> LoadXmlAsync<T>(string xml)
+	{
+		// Get or create the serializer for type T
+		var serializer = GetSerializer(typeof(T));
+
+		using var reader = new StringReader(xml);
+
+		// Deserialize synchronously since XmlSerializer doesn't support async operations
+		return (T)await Task.Run(() => serializer.Deserialize(reader)!);
+	}
+
+	private XmlSerializer GetSerializer(Type type)
+	{
+		if (!serializerCache.TryGetValue(type, out var serializer))
+		{
+			serializer = new XmlSerializer(type);
+			serializerCache[type] = serializer;
+		}
+		return serializer;
+	}
+
+
+	public async Task<SitesCommon_1.SiteList> GetAddonsListAsync()
+	{
+		var url = $"{GoogleAndroidRepositoryUrlBase}addons_list-6.xml";
```

Comment:
Do these numbers ever change? Are they loadable from somewhere?

Reply:
Yeah these are basically defined by the xsd / schemas built into the android sdk libs.   `addons_list-6.xml` basically derives from `sdk-sites-list-6.xsd`.  

There's not an index on the server (that I've found) which gives a comprehensive set of available xml files.  The only thing kinda like that is this addons_list-6.xml (and older versions) which does contain a list of addon xml files that can be parsed.

So we need to update this lib with new .xsd files if they are added to the android sdk manager sources ever.